includes/wikia/services/UserService.class.php

```diff
@@ -175,10 +175,23 @@ private function getUsersFromCacheByName( &$names ) {
 
 	private function getUserFromMemCacheById( $id ) {
 		$cacheIdKey = wfSharedMemcKey( "UserCache:".$id );
-		if ( ( $value = F::app()->wg->memc->get( $cacheIdKey ) ) !== false ) {
-			//cache locally
-			$this->cacheLocalUser( $value );
-			return $value;
+		$value = F::app()->wg->memc->get( $cacheIdKey );
+
+		if ( $value !== false ) {
+			if ( $value instanceof User ) {
+				//cache locally
+				$this->cacheLocalUser( $value );
```

Comment:
how about using try catch statement instead of instanceof here as type hinting was introduced for cacheLocalUser method?